File: chap6/play-sbook/app/comparison/PatternMatching.scala
package comparison

object PatternMatching {
    
    //204 chars
    def matchOnInt(value:Int) = 
        value match {
            case 1 => "one"
            case 2 => "two"
            case 3 => "three"
            case 4 => "four"
            case _ => "many"
        }


   





    //441 chars
    case class Data(i:Int, b:Boolean) {}
    def matchOnDomainObject(value:Data) =
        value match {
            case Data(1, true)            => "first"
            case Data(2, true)            => "second"
            case Data(3, true)            => "third"
            case Data(x, true) if x < 100 => "correct"
            case Data(_, true)            => "finished"
            case Data(_, false)           => "disqualified"
        }

}

File: chap3/play-jbook/app/comparison/PatternMatching.java
package comparison;

public class PatternMatching {
   
    //432 chars 
    public static String switchOnInt(int value) {
        String result = null;
        switch (value) {
            case 1: result = "one";
                    break;
            case 2: result = "two";
                    break;
            case 3: result = "three";
                    break;
            case 4: result = "four";
                    break;
            default: result = "many";
        }
        return result;
    }

    //715 chars
    public static class Data {
        public int i;
        public boolean b;

        public Data(int i, boolean b) {
            this.i = i;
            this.b = b;
        }
    }
    public static String checkDomainObject(Data data) {
        String result = null;
        if (!data.b) {
            result = "disqualified";
        } else {
            switch(data.i) {
                case 1: result = "first";
                        break;
                case 2: result = "second";
                        break;
                case 3: result = "third";
                        break;
                default: result = (data.i < 100)?"correct":"finished";
            }
        }
        return result;
    }

}